int32 state
string content

